climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,832 @@
+
+    in_geom = sh.vectorized.contains(poly_m, x_grid, y_grid)
+
+    if sum(in_geom.flatten()) > 1:
+        project_inv = pyproj.Transformer.from_proj(
+            pyproj.Proj("+proj=cea +lat_0=%f +lon_0=%f +units=m" %(lat_0, lon_0)),
+            pyproj.Proj(orig_crs),
+            always_xy=True
+        )
+        x_poly, y_poly = project_inv.transform(x_grid[in_geom], y_grid[in_geom])
+        return shgeom.MultiPoint(list(zip(x_poly, y_poly)))
+
+    LOGGER.warning('Polygon smaller than resolution. Setting a representative point.')
+    return shgeom.MultiPoint([repr_pnt])
+
+
+def poly_to_equalarea_proj(poly, orig_crs):
+    """
+    Project polyong to Equal Area Cylindrical projection
+    using a representative point as lat/lon reference.
+
+    https://proj.org/operations/projections/cea.html
+
+    Parameters
+    ----------
+    poly : shapely Polygon
+        Polygon
+    orig_crs: pyproj.CRS
+        CRS of the polygon
+
+    Returns
+    -------
+    poly : shapely Polygon
+        Polygon in equal are projection
+
+    """
+
+    repr_pnt = poly.representative_point()
+    lon_0, lat_0 = repr_pnt.x, repr_pnt.y
+
+    project = pyproj.Transformer.from_proj(
+        pyproj.Proj(orig_crs),
+        pyproj.Proj("+proj=cea +lat_0=%f +lon_0=%f +units=m" %(lat_0, lon_0)),
+        always_xy=True
+    )
+    return sh.ops.transform(project.transform, poly)
+
+def line_to_pnts(gdf_lines, res):
+
+    """ Convert a GeoDataframe with LineString geometries to
+    Point geometries, where Points are placed at a specified distance
+    along the original LineString. Each line is reduced to
+    at least two points.
+
+    Parameters
+    ----------
+    gdf_lines : gpd.GeoDataframe
+        Geodataframe with line geometries
+    res : float
+        Resolution (distance) apart from which the generated Points
+        should be approximately placed.
+
+    Returns
+    -------
+    gdf_points : gpd.GeoDataFrame
+        Geodataframe with a double index, first for line geometries,
+        second for the point disaggregation of the lines (i.e. one Point
+        per row).
+
+    See also
+    --------
+    * util.coordinates.compute_geodesic_lengths()
+    """
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf_lines.index.to_list() #To restore the naming of the index
+
+    gdf_points = gdf_lines.copy().reset_index(drop=True)
+    line_lengths = gdf_lines.length
+
+    line_fractions = [
+        np.linspace(0, 1, num=_pnts_per_line(length, res))
+        for length in line_lengths
+        ]
+
+    gdf_points['geometry_pnt'] = [
+        shgeom.MultiPoint([
+            line.interpolate(dist, normalized=True)
+            for dist in fractions
+            ])
+        for line, fractions in zip(gdf_points.geometry, line_fractions)
+        ]
+
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
```

Comment:
same question goes for the `poly_to_points`: is it necessary to store the polygon / line geometry  in the interpolated gdfs? I think this blows up the dataframe size quite a bit, but we don't really need it?

Reply:
In my opinion, I think the pros out way the cons. 1) Can you make some tests to see whether the size really increases? Because in general, the original geometries will be in a much smaller numbers than the points. Thus, I suspect that the extra memory needed would be only marginal (multi-index dataframes store their data intelligently as far as I know). 2) Not having both geometries in the same object means that external book-keeping must be made. This is in general very cumbersome and prone to errors. The only other solution I see would be to introduce another column with unique id numbers in order to be sure to know what point corresponds to what geometry. But I think this does not significantly reduce the memory requirements, and is vulnerable to errors. 3) In any case a user must keep both geometries somewhere. Thus, keeping it in the same, multi-index and efficient dataframe doing the book-keeping seems a good idea.

What do you think?

Reply:
Removed this method and changed the syntax a bit in the main impact_calc function, so the original geometries don't need to be carried over.

Reply:
What do you mean? It is always necessary to keep the original geometries...

Reply:
The point is: I think the dataframe should have both the original geometries and the point geometries, otherwise the risk for inconsistent data is very high as external book-keeping is required.

Reply:
ah, didn't see your earlier comment. Will check the memory part; the bookkeeping is indeed cumbersome unless using the top-level all-in-one function.

Reply:
Re-added it, since memory seems to not make a huuuuge difference.